pkg/model/resource/options_test.go

```diff
@@ -73,24 +76,37 @@ var _ = Describe("Resource Options", func() {
 			Expect(options.Validate().Error()).To(ContainSubstring("kind cannot be empty"))
 		})
 
-		It("should fail if the Kind is not pascal cased", func() {
-			// Base case
-			options := &Options{Group: "crew", Version: "v1", Kind: "FirstMate"}
-			Expect(options.Validate()).To(Succeed())
-
-			// Can't detect this case :(
-			options = &Options{Group: "crew", Version: "v1", Kind: "Firstmate"}
-			Expect(options.Validate()).To(Succeed())
-
-			options = &Options{Group: "crew", Version: "v1", Kind: "firstMate"}
-			Expect(options.Validate()).NotTo(Succeed())
-			Expect(options.Validate().Error()).To(ContainSubstring(
-				`kind must be PascalCase (expected FirstMate was firstMate)`))
+		DescribeTable("valid Kind values-according to core Kubernetes",
+			func(kind string) {
+				options := &Options{Group: "crew", Kind: kind, Version: "v1"}
+				Expect(options.Validate()).To(Succeed())
+			},
+			Entry("should pass validation if Kind is camelcase", "FirstMate"),
+			Entry("should pass validation if Kind has more than one caps at the start", "FIRSTMate"),
+		)
+
+		It("should fail if Kind is too long", func() {
+			kind := strings.Repeat("a", 64)
+
+			options := &Options{Group: "crew", Kind: kind, Version: "v1"}
+			err := options.Validate()
+			Expect(err).To(MatchError(ContainSubstring("must be no more than 63 characters")))
+		})
 
-			options = &Options{Group: "crew", Version: "v1", Kind: "firstmate"}
-			Expect(options.Validate()).NotTo(Succeed())
-			Expect(options.Validate().Error()).To(ContainSubstring(
-				`kind must be PascalCase (expected Firstmate was firstmate)`))
+		DescribeTable("invalid Kind values-according to core Kubernetes",
+			func(kind string) {
+				options := &Options{Group: "crew", Kind: kind, Version: "v1"}
+				Expect(options.Validate()).To(MatchError(
+					ContainSubstring("a DNS-1035 label must consist of lower case alphanumeric characters")))
+			},
+			Entry("should fail validation if Kind contains whitespaces", "Something withSpaces"),
+			Entry("should fail validation if Kind ends in -", "Hello-"),
```

Comment:
What about an extra `Entry("should fail validation if Kind starts with a number", "0Hello")`

Reply:
Good point, added one. Thanks!